### DISCUSSION TITLE
Will/Can there be an official Svelte port for Floating UI?

### DISCUSSION THREAD

**Discussion by Hugos68** [floating-ui/floating-ui#2834] [Ideas]
Svelte 5 is releasing stable soon and the maintainers have basically rewritten the reactivity model to use signals. The API is *so* simular I was able to port Floating UI's official Vue port easily to Svelte creating the following API:
```html
<script>
	import { useFloating, autoUpdate, flip, offset } from '@hugos68/svelte-floating-ui';

	const floating = useFloating();

	function style(node, style) {
	   const update = (new_style) => Object.assign(node.style, new_style);
	   update(style);
	   return { update };
	}
</script>

<button use:floating.reference>Button</button>
<div use:floating.floating use:style={floating.floatingStyles}>Tooltip</div>
```
See it in action: [REPL](https://svelte-5-preview.vercel.app/#H4sIAAAAAAAACmWQzU7DMBCEX2XlS1PJbW4IpUkEHLhyAE4EoTRZB6NkbdnroirKuyMnLS3iZs_MfvszCqV79CJ7GwXVA4pM3FsrpOCjjR9_wJ5RSOFNcE1Uct84bbmsqGI9WOMYRggeH3tTs6ZOQh3YvNq2ZpSgem0lGKU8MkygnBlgdfcZOuNvbtOFvlGn0k3Qq11FkdwY8gxnA4rrDsn6FFKBGtaGwPOxx4RMi3J5r2GMAVgwYR4GCkgIvz9OgaKEp_0XNrytvdcdzeXb2ZRwye1mzkJIriWHHBzF3Rf6NMtTRXl6uRDl-8BsKI6fnbfZOlTokBosH2Y3T5dUWVHe6sPf8O8NojoPUIz_zOeo-6l8MaZnbfO01YdSSDGYViuNrcjYBZzepx9jUSVQ8AEAAA==)

The API is basically identical to Vue except that the DOM references are obtained slightly differently.

Github: https://github.com/Hugos68/svelte-floating-ui

This is to showcase how creating the Svelte wrapper will soon be quite trivial to do, hence my question: 

Will/Can there be an official Svelte port for Floating UI?

**Comment by FezVrasta** [comment]
I'm not sure any of us maintainers have enough time or knowledge to maintain an official Svelte binding. I'm going to leave the final decision to @atomiks though.

**Reply by endigo9740** [reply]
Hi @FezVrasta just to provide some context here, Hugo and I work together on https://www.skeleton.dev/. I'm the creator and core maintainer for the project. The reason we're asking about Svelte support is because our library makes use for Floating UI for a number of popover elements, including: dropdown menus, tooltips, etc.

I wrote a simple [Svelte Action wrapper](https://github.com/skeletonlabs/skeleton/blob/master/packages/skeleton/src/lib/utilities/Popup/popup.ts) for basic Floating UI support in our previous versions of Skeleton, [which we detail here](https://www.skeleton.dev/utilities/popups). This was intentionally limited to provide ease of use, but of course our community is demanding a lot more!

We love Floating UI's [React hooks](https://floating-ui.com/docs/react), so we were hoping for something in the same ballpark for Svelte. Given what you mentioned above, we're considering taking this on ourselves as part of the [next major version of Skeleton](https://next.skeleton.dev/). We'll likely follow the hook model, per Hugo's suggestion in his post above.

Obviously we don't want to be a burden, but if you and your team are open to any form of collaboration, we would be down for that. We'll likely provide this as an open source package outside of Skeleton so anyone in the Svelte community can benefit.  If you're open to chatting about this please let me know and we can setup a time to meet and discuss further.

All that aside, we're all huge fans of Floating UI, so thanks for all the hard work! 👍

**Reply by endigo9740** [reply]
@FezVrasta just FYI, we're putting out this announcement tomorrow regarding our current progress and plans for a "Floating UI Svelte" - a dedicated Svelte 5 port of the Floating UI React package:

https://github.com/skeletonlabs/floating-ui-svelte/discussions/50

Again, if you're interested in working together, our door is always open!
